sherpa/data.py

```diff
@@ -488,10 +506,12 @@ def apply(self, array):
         if self.mask is False:
             raise DataErr('notmask')
 
+        # Ensure we always return a ndarray.
+        #
+        array = numpy.asarray(array)
```

Comment:
In a later check we wend up passing `array` to `_check`, which calls `npumpy.asanyarray`. In general we have a "passing regard" for numpy subclasses, as there's a number of places we just call `asarray` when we perhaps could use `asanyarray`. It would need some tests set up so we could do this properly, which is definitely a problem for another year ;-)